### ISSUE TITLE
Some editor settings are not being respected by language

### ISSUE THREAD

**Issue by sandy081** [Microsoft/vscode#19511]
Following settings are not getting update by language

```
editor.tabSize
editor.insertSpaces
editor.detectIndentation
editor.trimAutoWhitespace
```

**Comment by sandy081** [comment]
Above settings comes from the Model and are created/updated by the modelService. Fixing this need adoption in modelService to language based changes to configurations.

**Comment by sandy081** [comment]
Current code is not ready to provide a simple fix for this. This needs quiet amount of refactoring
- Model service has to be adopted to language based settings
- Model builder and its callers has to be refactored and adopted to language based settings. Core of the problem is that Model builder consumes config options to create `IRawText` which is an input for creating a Model. Since, config options are independent of language there were no issues. But from now, they are dependent on language. To get language we either need file URI and file first line. First line is not available until Model Builder builds IRawText. So it is a chicken-egg problem which needs proper split of `IRawText` to `IConfigurationUnawareRawText`. And, Model builder spits this new `IConfigurationUnawareRawText` and the Model Service can create the model from this `IConfigurationUnawareRawText` and the language.

This from an implementation perspective a feature by itself and is risky to implement this at the very end of 1.9. Hence we decided to revert this fix and make this available in February release. 

So long story short, in 1.9 release, users cannot configure above described settings by language. This will be available in latest insiders and from next stable release.

**Comment by mdalpi** [comment]
Maybe `files.exclude` should be added to the list, I'm getting _"Unknown identifier. Use language identifiers"_ trying to set under Python settings.

**Comment by sandy081** [comment]
@mdalpi Please let me know the use case behind supporting `files.exclude` in the list?  Also please create a separate issue for that. Thanks.

**Comment by mdalpi** [comment]
@sandy081 I'll use just to ignore cache files (`*.pyc` files and `__pycache__` folder specifically). Of course I can do with the global setting just fine, I just thought that this would be possible. We should wait for others to create an issue about this (maybe I'm the only trying this particular case). Thanks.

**Comment by SimplyPhy** [comment]
I just ran into this issue was well.  I tried to set tabsize to 4 for scss, and 2 for js.  I now know why it doesn't work.  That said, if the detection algorithm worked (doesn't work in my scss for unknown reason; all tabs were 4, and it defaults to 2), I wouldn't be concerned by this issue.

**Comment by kristianmandrup** [comment]
I'm currently unable to avoid auto-save indentation of 4 spaces for my `.js` files.
I tried everything. Disabled all linter extensions. Configuring `.vscode/settings.json`

```json
{
    "vsicons.presets.angular": false,
    "editor.tabSize": 2,
    "editor.detectIndentation": false,
    "editor.formatOnSave": false
}
```

Added the same for my `preferences/user settings` file:

```json
// Place your settings in this file to overwrite the default settings
{
    "editor.tabSize": 2,
    "editor.detectIndentation": false,
    "editor.formatOnSave": false,
    "typescript.check.tscVersion": false,
    "window.zoomLevel": 0,
    "files.trimTrailingWhitespace": true,
    "npm.validate.enable": false,
    "editor.minimap.enabled": true,
    "workbench.iconTheme": "vscode-icons",
    "[javascript]": {
        "editor.tabSize": 2
    }
}
```

Still it auto-indents on every save to 4 spaces! Is there any way to get around this on the current release? Or is it some extension playing tricks on me?

**Comment by sandy081** [comment]
@kristianmandrup You can try disabling all extensions, with vanilla VS Code, do following

- Have empty `.vscode/settings.json`
- In user settings have following 
```
"editor.detecetIndentation": false,
"editor.tabSize": 2
```
Above user settings will have tab size to 2 for all files. To do just for javascript files have following in user settings
```
"[javascript]": {
"editor.detecetIndentation": false,
"editor.tabSize": 2
}
```

Let me know if above works for you?

**Comment by kristianmandrup** [comment]
This is exactly what I tried (see above), except disabling all my extensions. Thanks anyways.

**Comment by sandy081** [comment]
can you try disabling extensions and see if it works?

**Comment by kristianmandrup** [comment]
Turned out the [vscode-JS-CSS-HTML-formatter extension](https://github.com/Lonefy/vscode-JS-CSS-HTML-formatter) was to blame. Solved now! Thanks

**Comment by ryenus** [comment]
I just encountered the same trap, and @kristianmandrup is right. 
For those installed [JS-CSS-HTML Formatter][1] extension, please check **Formatter Config** (via Command Palette), especially, the following (default) settings:

	"javascript": {
	  "indent_size": 4,
	  "end_with_newline": true
	  // ...
	}


[1]: https://github.com/Lonefy/vscode-JS-CSS-HTML-formatter

**Comment by Earl-Brown** [comment]
@sandy081 - the editor.detectIndentation trick worked for me.  Thanks!
